core-blocks/file/index.js

```diff
@@ -123,6 +132,14 @@ export const settings = {
 			{
 				type: 'block',
 				blocks: [ 'core/video' ],
+				canTransform: ( { id } ) => {
+					if ( ! id ) {
+						return false;
+					}
+					const { getMedia } = select( 'core' );
+					const media = getMedia( id );
+					return media && media.mime_type.indexOf( 'video' ) > -1;
```

Comment:
first stupid nit: If `media` is `null` then this would return `null` instead of the expected `false`. We generally cast to a boolean to address this.

```js
return !! media && media.mime_type.indexOf( 'video' ) > -1;
```

second stupid nit: We could use `includes` here.

```js
return !! media && includes( media.mime_type, 'video' );
```